pkg/deploy/lattice/service_network_manager.go

```diff
@@ -53,66 +57,72 @@ type defaultServiceNetworkManager struct {
 
-		service_networkInput := vpclattice.CreateServiceNetworkInput{
-			Name: &service_network.Spec.Name,
+		serviceNetworkInput := vpclattice.CreateServiceNetworkInput{
+			Name: &stackSn.Spec.Name,
 			Tags: make(map[string]*string),
 		}
-		service_networkInput.Tags[latticemodel.K8SServiceNetworkOwnedByVPC] = &config.VpcID
+		serviceNetworkInput.Tags[latticemodel.K8SServiceNetworkOwnedByVPC] = &config.VpcID
 
-		glog.V(2).Infof("Create service_network >>>> req[%v]", service_networkInput)
-		resp, err := vpcLatticeSess.CreateServiceNetworkWithContext(ctx, &service_networkInput)
-		glog.V(2).Infof("Create service_network >>>> resp[%v], err : %v", resp, err)
+		glog.V(2).Infof("CreateOrUpdate stackSn >>>> req[%v]", serviceNetworkInput)
+		resp, err := vpcLatticeSess.CreateServiceNetworkWithContext(ctx, &serviceNetworkInput)
+		glog.V(2).Infof("CreateOrUpdate stackSn >>>> resp[%v], err : %v", resp, err)
 		if err != nil {
-			glog.V(2).Infof("Failed to create service_network[%v], err: %v", service_network, err)
+			glog.V(2).Infof("Failed to create stackSn[%v], err: %v", stackSn, err)
 			return latticemodel.ServiceNetworkStatus{ServiceNetworkARN: "", ServiceNetworkID: ""}, err
 		}
 		service_networkID = aws.StringValue(resp.Id)
 		service_networkArn = aws.StringValue(resp.Arn)
-		isServiceNetworkAssociatedWithVPC = false
-		glog.V(6).Infof(" ServiceNetwork Create API resp [%v]\n", resp)
+		isServiceNetworkAlreadyAssociatedWithCurrentVpc = false
+		glog.V(6).Infof(" ServiceNetwork CreateOrUpdate API resp [%v]\n", resp)
 
-	} else {
-		glog.V(6).Infof("service_network[%v] exists, further check association", service_network)
+	} else { //service_networkSummary != nil
+		glog.V(6).Infof("stackSn[%v] exists, further check association", stackSn)
 		service_networkID = aws.StringValue(service_networkSummary.SvcNetwork.Id)
 		service_networkArn = aws.StringValue(service_networkSummary.SvcNetwork.Arn)
-		isServiceNetworkAssociatedWithVPC, service_networkAssociatedWithCurrentVPCId, _, err = m.isServiceNetworkAssociatedWithVPC(ctx, service_networkID)
+		isServiceNetworkAlreadyAssociatedWithCurrentVpc, snvaAssociatedWithCurrentVPC, _, err = m.isServiceNetworkAlreadyAssociatedWithVPC(ctx, service_networkID)
 		if err != nil {
 			return latticemodel.ServiceNetworkStatus{ServiceNetworkARN: "", ServiceNetworkID: ""}, err
 		}
+		if stackSn.Spec.AssociateToVPC == true && isServiceNetworkAlreadyAssociatedWithCurrentVpc == true &&
+			snvaAssociatedWithCurrentVPC.Status != nil && aws.StringValue(snvaAssociatedWithCurrentVPC.Status) == vpclattice.ServiceNetworkVpcAssociationStatusActive {
+			return m.UpdateServiceNetworkVpcAssociationIfNeeded(ctx, service_networkID, service_networkArn, stackSn, snvaAssociatedWithCurrentVPC.Id)
+
+		}
 	}
 
-	if service_network.Spec.AssociateToVPC == true {
-		if isServiceNetworkAssociatedWithVPC == false {
+	if stackSn.Spec.AssociateToVPC == true {
+		if isServiceNetworkAlreadyAssociatedWithCurrentVpc == false {
```

Comment:
nit: you can use the short version of the condition in if statements. like `if condition` or `if !condition`